### PR TITLE
fix: Merge & Git Status UI

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

Add DeployedAt UI and merge status component

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/git_status_msg_ui 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.6 **(0.01)** | 36.43 **(-0.02)** | 33.22 **(0)** | 55.12 **(0.01)**
 :red_circle: | app/client/src/actions/gitSyncActions.ts | 63.55 **(-0.09)** | 100 **(0)** | 0 **(0)** | 76.09 **(-0.51)**
 :red_circle: | app/client/src/api/GitSyncAPI.tsx | 41.3 **(-0.37)** | 100 **(0)** | 7.14 **(0.47)** | 41.86 **(-0.36)**
 :red_circle: | app/client/src/constants/messages.ts | 73.99 **(-0.05)** | 100 **(0)** | 23.06 **(-0.14)** | 79.88 **(0.09)**
 :green_circle: | app/client/src/reducers/uiReducers/gitSyncReducer.ts | 15.38 **(0.75)** | 100 **(0)** | 0 **(0)** | 15.38 **(0.75)**
 :red_circle: | app/client/src/sagas/GitSyncSagas.ts | 11.04 **(-0.39)** | 1.47 **(0.04)** | 7.14 **(0.47)** | 14.63 **(-0.52)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :red_circle: | app/client/src/selectors/gitSyncSelectors.tsx | 65.88 **(-0.79)** | 30 **(-7.5)** | 6.67 **(-0.47)** | 56.9 **(0.54)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.04 **(-0.22)** | 40.42 **(-0.83)** | 34.48 **(0)** | 55.96 **(-0.26)**</details>

### CODE REVIEW

File: app/client/src/reducers/uiReducers/gitSyncReducer.ts [hunk 1]
```diff
@@ -277,21 +279,23 @@ export type GitSyncReducerState = {
   isCommitSuccessful: boolean;
   isPushSuccessful: boolean;
   isPushingToGit?: boolean;
+
+  fetchingBranches: boolean;
+  isFetchingGitConfig: boolean;
+  isFetchingLocalGitConfig: boolean;
+  isDisconnectingGit: boolean;
```
Comment (rishabhsaxena) [inline app/client/src/reducers/uiReducers/gitSyncReducer.ts:286]: nit: I guess we're not using `isDisconnectingGit`

**Review by rishabhsaxena — APPROVED** [review]
LGTM!

File: app/client/src/assets/svg/error-warning-line.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M7.99967 14.6663C4.31767 14.6663 1.33301 11.6817 1.33301 7.99967C1.33301 4.31767 4.31767 1.33301 7.99967 1.33301C11.6817 1.33301 14.6663 4.31768 14.6663 7.99967C14.6663 11.6817 11.6817 14.6663 7.99967 14.6663ZM7.99967 13.333C9.41416 13.333 10.7707 12.7711 11.7709 11.7709C12.7711 10.7707 13.333 9.41416 13.333 7.99967C13.333 6.58519 12.7711 5.22863 11.7709 4.22844C10.7707 3.22824 9.41416 2.66634 7.99967 2.66634C6.58519 2.66634 5.22863 3.22824 4.22844 4.22844C3.22824 5.22863 2.66634 6.58519 2.66634 7.99967C2.66634 9.41416 3.22824 10.7707 4.22844 11.7709C5.22863 12.7711 6.58519 13.333 7.99967 13.333ZM7.33301 9.99967L8.66634 9.99967L8.66634 11.333L7.33301 11.333L7.33301 9.99967ZM7.33301 4.66634L8.66634 4.66634L8.66634 8.66634L7.33301 8.66634L7.33301 4.66634Z" fill="#D71010"/>
+</svg>
```
Comment (haojin111) [inline app/client/src/assets/svg/error-warning-line.svg:3]: Can you use a similar icon instead of svg directly?

**Review by rishabhsaxena — DISMISSED** [review]

**Pull request merged**